internal/controller/apiextensions/claim/syncer_ssa.go

```diff
@@ -0,0 +1,292 @@
+			return nil
+		}
+	}
+	p, err := csaupgrade.UpgradeManagedFieldsPatch(obj, sets.New[string](csaManagers...), ssaManager)
+	if err != nil {
+		return errors.Wrap(err, errCreatePatch)
+	}
+	if p == nil {
+		// No patch means there's nothing to upgrade.
+		return nil
+	}
+	return errors.Wrap(resource.IgnoreNotFound(u.client.Patch(ctx, obj, client.RawPatch(types.JSONPatchType, p))), errPatchFieldManagers)
+}
+
+// A ServerSideCompositeSyncer binds and syncs a claim with a composite resource
+// (XR). It uses server-side apply to update the XR.
+type ServerSideCompositeSyncer struct {
+	client client.Client
+	names  names.NameGenerator
+}
+
+// NewServerSideCompositeSyncer returns a CompositeSyncer that uses server-side
+// apply to sync a claim with a composite resource.
+func NewServerSideCompositeSyncer(c client.Client, ng names.NameGenerator) *ServerSideCompositeSyncer {
+	return &ServerSideCompositeSyncer{client: c, names: ng}
+}
+
+// Sync the supplied claim with the supplied composite resource (XR). Syncing
+// may involve creating and binding the XR.
+func (s *ServerSideCompositeSyncer) Sync(ctx context.Context, cm *claim.Unstructured, xr *composite.Unstructured) error { //nolint:gocyclo // This complex process seems easier to follow in one long method.
+	// First we sync claim -> XR.
+
+	// Create an empty XR patch object. We'll use this object to ensure we only
+	// SSA our desired state, not the state we previously read from the API
+	// server.
+	xrPatch := composite.New(composite.WithGroupVersionKind(xr.GroupVersionKind()))
+
+	// If the claim references an XR, make sure we're going to apply that XR. We
+	// do this instead of using the supplied XR's name just in case the XR
+	// exists, but we couldn't get it due to a stale cache.
+	if ref := cm.GetResourceReference(); ref != nil {
+		xrPatch.SetName(ref.Name)
+	}
+
+	// If the XR doesn't have a name (i.e. doesn't exist), derive a name from
+	// the claim. The generated name is likely (but not guaranteed) to be
+	// available when we create the XR. If taken, then we are going to update an
+	// existing XR, probably hijacking it from another claim.
+	if xrPatch.GetName() == "" {
+		xrPatch.SetGenerateName(fmt.Sprintf("%s-", cm.GetName()))
+		if err := s.names.GenerateName(ctx, xrPatch); err != nil {
+			return errors.Wrap(err, errGenerateName)
+		}
+	}
+
+	// It's possible we're being asked to configure a statically provisioned XR.
+	// We should respect its existing external name annotation.
+	en := meta.GetExternalName(xr)
+
+	// Do not propagate *.kubernetes.io annotations/labels from claim to XR. For
+	// example kubectl.kubernetes.io/last-applied-configuration should not be
+	// propagated.
+	// See https://kubernetes.io/docs/reference/labels-annotations-taints/
+	// for all annotations and their semantic
+	if ann := withoutReservedK8sEntries(cm.GetAnnotations()); len(ann) > 0 {
+		meta.AddAnnotations(xrPatch, withoutReservedK8sEntries(cm.GetAnnotations()))
+	}
```

Comment:
Is this slightly different w.r.t. the syncer_csa.go equivalent for a reason?
It's just as below there:
```
	meta.AddAnnotations(xr, withoutReservedK8sEntries(cm.GetAnnotations()))
```

Reply:
It looks like @pedjak made this change.

See current SSA-based implementation:

https://github.com/crossplane/crossplane/blob/6f758cec20318f148c4f8a902dfde0a5c9d2c81d/internal/controller/apiextensions/claim/configurator.go#L90

Previous CSA based implementation:

https://github.com/crossplane/crossplane/blob/13c2d1f42c61d332b065f3b20c1db6e10c773c37/internal/controller/apiextensions/claim/configurator.go#L102

My guess is we do this to avoid explicitly deleting all annotations from the XR, even annotations that this claim didn't set. I think:

* If there are no annotations `cm.GetAnnotations()` returns `map[string]string{}`.
* We then call `xrPatch.SetAnnotations(map[string]string{})`
* This results in a patch that appears to set `metadata.annotations: {}`
* SSA interprets that as "I want there to be no annotations on the XR - remove them all".

Whereas I think if we just omit `metadata.annotations` completely from the patch it has a slightly different meaning - "I want to delete any annotations _this field manager_ previously created".